editor/editor_node.cpp

```diff
@@ -3126,6 +3121,86 @@ void EditorNode::stop_child_process() {
 	_menu_option_confirm(RUN_STOP, false);
 }
 
+Ref<Texture> EditorNode::get_object_icon(const Object *p_object, const String &p_fallback) const {
+	ERR_FAIL_COND_V(!p_object || !gui_base, NULL);
+
+	Ref<Script> script = p_object->get_script();
+	if (script.is_null() && p_object->is_class("Script")) {
+		script = p_object;
+	}
+
+	StringName name;
+	String icon_path;
+	if (script.is_valid()) {
+		name = EditorNode::get_editor_data().script_class_get_name(script->get_path());
+		icon_path = EditorNode::get_editor_data().script_class_get_icon_path(name);
+		name = script->get_instance_base_type();
+	}
+
+	if (gui_base->has_icon(p_object->get_class(), "EditorIcons"))
+		return gui_base->get_icon(p_object->get_class(), "EditorIcons");
+
+	if (icon_path.length())
+		return ResourceLoader::load(icon_path);
+
+	if (p_object->has_meta("_editor_icon"))
+		return p_object->get_meta("_editor_icon");
+
+	if (name != StringName()) {
+		const Map<String, Vector<EditorData::CustomType> > &p_map = EditorNode::get_editor_data().get_custom_types();
+		for (const Map<String, Vector<EditorData::CustomType> >::Element *E = p_map.front(); E; E = E->next()) {
+			const Vector<EditorData::CustomType> &ct = E->value();
+			for (int i = 0; i < ct.size(); ++i) {
+				if (ct[i].name == name && ct[i].icon.is_valid()) {
+					return ct[i].icon;
+				}
+			}
+		}
+	}
+
+	if (p_fallback.length())
+		return gui_base->get_icon(p_fallback, "EditorIcons");
+
+	return NULL;
+}
+
+Ref<Texture> EditorNode::get_object_icon_name(const String &p_class, const String &p_fallback) const {
```

Comment:
Two suggestions about this function:
* rename to get_class_icon (i think it makes more sense)
* fallback could have a default argument "Object" so it does not need to be passed always

Reply:
Done.